tests/test_taskFactory.py

```diff
@@ -0,0 +1,162 @@
+# This file is part of ctrl_mpexec.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import shutil
+import tempfile
+import unittest
+
+import lsst.afw.table as afwTable
+import lsst.daf.butler.tests as butlerTests
+import lsst.pex.config as pexConfig
+from lsst.pipe.base import PipelineTaskConfig, PipelineTaskConnections
+from lsst.pipe.base.configOverrides import ConfigOverrides
+from lsst.pipe.base import connectionTypes
+
+from lsst.ctrl.mpexec import TaskFactory
+
+
+class FakeConnections(PipelineTaskConnections, dimensions=set()):
+    initInput = connectionTypes.InitInput(name="fakeInitInput", doc="", storageClass="SourceCatalog")
+    initOutput = connectionTypes.InitOutput(name="fakeInitOutput", doc="", storageClass="SourceCatalog")
+    input = connectionTypes.Input(name="fakeInput", doc="", storageClass="SourceCatalog", dimensions=set())
+    output = connectionTypes.Output(name="fakeOutput", doc="", storageClass="SourceCatalog", dimensions=set())
+
+
+class FakeConfig(PipelineTaskConfig, pipelineConnections=FakeConnections):
+    widget = pexConfig.Field(dtype=float, doc="")
+
+
+def mockTaskClass():
+    """A class placeholder that records calls to __call__.
+    """
+    mock = unittest.mock.Mock(__name__="_TaskMock", _DefaultName="FakeTask", ConfigClass=FakeConfig)
+    return mock
+
+
+class TaskFactoryTestCase(unittest.TestCase):
```

Comment:
Please add a docstring: it's not obvious to me what a "TaskFactory" test would be testing. I think it's testing `TaskFactory.makeTask()` with a variety of combinations of optional arguments?

Reply:
Hmmm, maybe the fact that `TaskFactory` itself is short on documentation is really my problem? I was looking at the test without knowing anything about TaskFactory: I didn't know it only had one method.